File: frontend/src/microservices/flight/objects/BookFlightPlannerRequest.ts
// 本文件定义 flight 模块的 `BookFlightPlannerRequest`，作为 planner 请求参数并提供 JSON 编解码。

export type BookFlightPlannerRequest = {
  userId: string
  flightId: string
  travelerIds: string[]
  cabinClass: string
}

export const bookFlightPlannerRequestFromJson = (json: string): BookFlightPlannerRequest =>
  JSON.parse(json) as BookFlightPlannerRequest

export const bookFlightPlannerRequestToJson = (value: BookFlightPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/BookFlightPlannerRequest.scala
// BookFlightPlannerRequest defines the flight booking planner request model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BookFlightPlannerRequest(
    userId: String,
    flightId: String,
    travelerIds: List[String] = Nil,
    cabinClass: String
)

object BookFlightPlannerRequest:
  given sourceEncoder: Encoder[BookFlightPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BookFlightPlannerRequest] = deriveDecoder

